public/js/index.js

```diff
@@ -1935,6 +1935,78 @@ function u2fRegisterRequest() {
     });
 }
 
+function createWorker(fn, ...args) {
+    const argsString = args.map(arg => JSON.stringify(arg)).join(", ");
+    const url = URL.createObjectURL(new Blob([`(${fn})(${argsString})`], {type: "application/javascript"})),
+    worker = new Worker(url);
+    URL.revokeObjectURL(url);
+    return worker;
+}
+
+function randomString() {
+    return Math.random().toString(36).substring(7);
+}
+
+function createHighlightWorker() {
+    if (!window.config || !window.config.hljsUrl) return;
+
+    // convert relative URL to absolute for importScripts to accept it
+    let url;
+    try {
+        url = new URL(window.config.hljsUrl, window.location.origin).href;
+    } catch (err) {}
+    if (!url) return;
+
+    window.highlightWorker = createWorker(url => {
+        self.importScripts(url);
+        self.onmessage = ({data}) => {
+            const {id, texts} = data;
+            if (!id || !texts || !texts.length) return;
+
+            const results = texts.map(text => {
+                return (self.hljs.highlightAuto(text) || {}).value;
+            });
+            self.postMessage({id, results});
+        }
+    }, url);
+}
+
+function highlight(nodes) {
+    if (!window.highlightWorker) return;
+
+    nodes = [].slice.call(nodes || []);
+    if (!nodes.length) return;
+
+    // requestId is used to identify this particular highlight request so we can prevent race conditions
+    // when parallel highlighting requests do not finish in the order they were sent in.
+    const requestId = randomString();
+
+    function listener({data}) {
+        const {id, results} = data;
+        if (id !== requestId) return;
+
+        window.highlightWorker.removeEventListener("message", listener);
+
+        for (const [index, result] of Object.entries(results)) {
+            const node = nodes[index];
+
+            // highlight.js in a web worker does not accept HTML strings so we have to pass .textContent to it
+            // and replicate what the golang templating does by re-adding these ol > li wrappings where necessary.
+            if (node.querySelector("ol.linenums")) {
+                const lines = result.split(/\r?\n/).map((line, i) => `<li class="L${i + 1}" rel="L${i + 1}">${line}</li>`);
+                node.classList.add("hljs");
+                node.innerHTML = `<ol class="linenums">${lines.join("")}<ol>`;
```

Comment:
This `ol.linenums` business seems to only be present in rendered files to support adding the yellow background on linked lines. Not happy having to do this wrapping here at all. I think future improvlement would be to eliminate these children of `pre > code` so that the output of the highlighter can be rendered as-is in all cases.

Reply:
Related: https://github.com/highlightjs/highlight.js/issues/2086